modules/rostests/apitests/user32/SW_FORCEMINIMIZE.c

```diff
@@ -0,0 +1,163 @@
+
+    SetWindowLongPtrA(hwnd, GWL_EXSTYLE, WS_EX_MAKEVISIBLEWHENUNGHOSTED);
+    ret = ShowWindow(hwnd, SW_FORCEMINIMIZE);
+    Sleep(100);
+    ok(ret != 0, "ret was FALSE");
+    ok(CheckColor() != RED, "Color was red\n");
+    style = GetWindowLongPtrA(hwnd, GWL_STYLE);
+    if (IsWindowsVistaOrGreater())
+        ok((style & (WS_POPUP | WS_MINIMIZE | WS_VISIBLE)) == (WS_POPUP | WS_MINIMIZE | WS_VISIBLE), "style was 0x%08lX", style);
+    else
+        ok((style & (WS_POPUP | WS_MINIMIZE | WS_VISIBLE)) == (WS_POPUP), "style was 0x%08lX", style);
+    exstyle = GetWindowLongPtrA(hwnd, GWL_EXSTYLE);
+    ok_long(exstyle, 0);
+
+    ShowWindow(hwnd, SW_MINIMIZE);
+    Sleep(100);
+    ok(CheckColor() != RED, "Color was red\n");
+
+    SetWindowLongPtrA(hwnd, GWL_EXSTYLE, WS_EX_MAKEVISIBLEWHENUNGHOSTED);
+    ret = ShowWindow(hwnd, SW_FORCEMINIMIZE);
+    Sleep(100);
+    ok(ret != 0, "ret was FALSE");
+    ok(CheckColor() != RED, "Color was red\n");
+    style = GetWindowLongPtrA(hwnd, GWL_STYLE);
+    ok((style & (WS_POPUP | WS_MINIMIZE | WS_VISIBLE)) == (WS_POPUP | WS_MINIMIZE | WS_VISIBLE), "style was 0x%08lX", style);
+    exstyle = GetWindowLongPtrA(hwnd, GWL_EXSTYLE);
+    ok_long(exstyle, 0);
+
+    ShowWindow(hwnd, SW_SHOWNORMAL);
+    Sleep(100);
+    ok_long(CheckColor(), RED);
+
+    SetWindowLongPtrA(hwnd, GWL_EXSTYLE, WS_EX_MAKEVISIBLEWHENUNGHOSTED);
+    ret = ShowWindow(hwnd, SW_FORCEMINIMIZE);
+    Sleep(100);
+    ok(ret != 0, "ret was FALSE");
+    ok(CheckColor() != RED, "Color was red\n");
+    style = GetWindowLongPtrA(hwnd, GWL_STYLE);
+    if (IsWindowsVistaOrGreater())
+        ok((style & (WS_POPUP | WS_MINIMIZE | WS_VISIBLE)) == (WS_POPUP | WS_MINIMIZE | WS_VISIBLE), "style was 0x%08lX", style);
+    else
+        ok((style & (WS_POPUP | WS_MINIMIZE | WS_VISIBLE)) == (WS_POPUP), "style was 0x%08lX", style);
+    exstyle = GetWindowLongPtrA(hwnd, GWL_EXSTYLE);
+    ok_long(exstyle, 0);
+
+    SetWindowLongPtrA(hwnd, GWL_EXSTYLE, WS_EX_MAKEVISIBLEWHENUNGHOSTED);
+    ret = ShowWindow(hwnd, SW_FORCEMINIMIZE);
+    Sleep(100);
+    if (IsWindowsVistaOrGreater())
+        ok(ret, "ret was FALSE");
+    else
+        ok(!ret, "ret was TRUE");
+    ok(CheckColor() != RED, "Color was red\n");
+    style = GetWindowLongPtrA(hwnd, GWL_STYLE);
+    if (IsWindowsVistaOrGreater())
+        ok((style & (WS_POPUP | WS_MINIMIZE | WS_VISIBLE)) == (WS_POPUP | WS_MINIMIZE | WS_VISIBLE), "style was 0x%08lX", style);
+    else
+        ok((style & (WS_POPUP | WS_MINIMIZE | WS_VISIBLE)) == (WS_POPUP), "style was 0x%08lX", style);
+    exstyle = GetWindowLongPtrA(hwnd, GWL_EXSTYLE);
+    ok_long(exstyle, 0);
+
+    PostMessageA(hwnd, WM_CLOSE, 0, 0);
+    return 0;
+}
+
+static LRESULT CALLBACK
+WindowProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+    switch (uMsg)
+    {
+        case WM_CREATE:
+        {
+            DWORD dwThreadId;
+            g_hThread = CreateThread(NULL, 0, ThreadFunc, hwnd, 0, &dwThreadId);
```

Comment:
What about creating and starting the thread in `Test_WS_FORCEMINIMIZE` instead? :thinking:

Reply:
Moved.